tests/gateway/providers/test_clarifai.py

```diff
@@ -0,0 +1,420 @@
+                        -0.004586036,
+                        0.009094779,
+                        -0.010943364,
+                    ],
+                    "index": 0,
+                }
+            ],
+            "model": "multimodal-clip-embed",
+            "usage": {"prompt_tokens": None, "total_tokens": None},
+        }
+        mock_post.assert_called_once_with(
+            "https://api.clarifai.com/v2/users/clarifai/apps/main/models/multimodal-clip-embed/outputs",
+            json={
+                "inputs": [
+                    {
+                        "data": {
+                            "text": {
+                                "raw": "hello world",
+                            }
+                        }
+                    }
+                ],
+            },
+            timeout=ClientTimeout(total=MLFLOW_GATEWAY_ROUTE_TIMEOUT_SECONDS),
+        )
+
+
+@pytest.mark.asyncio
+async def test_embeddings_batch():
+    resp = embeddings_batch_response()
+    config = embeddings_config()
+    with mock.patch(
+        "aiohttp.ClientSession.post", return_value=MockAsyncResponse(resp)
+    ) as mock_post:
+        provider = ClarifaiProvider(RouteConfig(**config))
+        payload = {
+            "input": ["hello world", "bye world"],
+        }
+        response = await provider.embeddings(embeddings.RequestPayload(**payload))
+        assert jsonable_encoder(response) == {
+            "object": "list",
+            "data": [
+                {
+                    "object": "embedding",
+                    "embedding": [
+                        0.008268436,
+                        0.010364032,
+                        -0.013782379,
+                    ],
+                    "index": 0,
+                },
+                {
+                    "object": "embedding",
+                    "embedding": [
+                        0.017018614,
+                        0.016557906,
+                        -0.00271358,
+                    ],
+                    "index": 1,
+                },
+            ],
+            "model": "multimodal-clip-embed",
+            "usage": {"prompt_tokens": None, "total_tokens": None},
+        }
+        mock_post.assert_called_once_with(
+            "https://api.clarifai.com/v2/users/clarifai/apps/main/models/multimodal-clip-embed/outputs",
+            json={
+                "inputs": [
+                    {
+                        "data": {
+                            "text": {
+                                "raw": "hello world",
+                            }
+                        }
+                    },
+                    {
+                        "data": {
+                            "text": {
+                                "raw": "bye world",
+                            }
+                        }
+                    },
+                ],
+            },
+            timeout=ClientTimeout(total=MLFLOW_GATEWAY_ROUTE_TIMEOUT_SECONDS),
+        )
+
```

Comment:
Can we add a quick validation that the error handling functions as expected for the chat endpoint raising a 422 with the appropriate text response?